src/vs/workbench/contrib/terminalContrib/suggest/test/browser/terminalCompletionService.test.ts

```diff
@@ -67,129 +95,247 @@ suite('TerminalCompletionService', () => {
-				isFile: false,
-				replacementIndex: 3,
-				replacementLength: 0
-			}]);
-		});
-		test('cd .|', async () => {
+			const result = await terminalCompletionService.resolveResources(resourceRequestConfig, 'cd ./f', 6, provider);
+
+			assertCompletions(result, [
+				{ label: './', detail: 'test' },
+				{ label: './folder1/' },
+				{ label: './../' },
+			], { replacementIndex: 3, replacementLength: 3 });
+		});
+	});
+
+	suite('resolveResources should handle file and folder completion requests correctly', () => {
+		setup(() => {
+			validResources = [URI.parse('file:///test')];
+			childResources = [
+				{ resource: URI.parse('file:///test/.hiddenFile'), isFile: true },
+				{ resource: URI.parse('file:///test/.hiddenFolder/'), isDirectory: true },
+				{ resource: URI.parse('file:///test/folder1/'), isDirectory: true },
+				{ resource: URI.parse('file:///test/file1.txt'), isFile: true },
+			];
+		});
+
+		test('./| should handle hidden files and folders', async () => {
 			const resourceRequestConfig: TerminalResourceRequestConfig = {
 				cwd: URI.parse('file:///test'),
 				foldersRequested: true,
+				filesRequested: true,
 				pathSeparator
 			};
-			const result = await terminalCompletionService.resolveResources(resourceRequestConfig, 'cd .', 4, provider);
-			assert.deepEqual(result, [{
-				label: `.${pathSeparator}folder1${pathSeparator}`,
-				provider,
-				kind: TerminalCompletionItemKind.Folder,
-				isDirectory: true,
-				isFile: false,
-				replacementIndex: 3,
-				replacementLength: 1 // replacing .
-			}]);
-		});
-		test('cd ./|', async () => {
+			const result = await terminalCompletionService.resolveResources(resourceRequestConfig, './', 2, provider);
+
+			assertCompletions(result, [
+				{ label: './', detail: 'test' },
+				{ label: './.hiddenFile', kind: TerminalCompletionItemKind.File },
+				{ label: './.hiddenFolder/', kind: TerminalCompletionItemKind.Folder },
+				{ label: './folder1/', kind: TerminalCompletionItemKind.Folder },
+				{ label: './file1.txt', kind: TerminalCompletionItemKind.File },
+				{ label: './../', kind: TerminalCompletionItemKind.Folder },
+			], { replacementIndex: 0, replacementLength: 2 });
+		});
+
+		test('./h| should handle hidden files and folders', async () => {
 			const resourceRequestConfig: TerminalResourceRequestConfig = {
 				cwd: URI.parse('file:///test'),
 				foldersRequested: true,
+				filesRequested: true,
 				pathSeparator
 			};
-			const result = await terminalCompletionService.resolveResources(resourceRequestConfig, 'cd ./', 5, provider);
-			assert.deepEqual(result, [{
-				label: `.${pathSeparator}folder1${pathSeparator}`,
-				provider,
-				kind: TerminalCompletionItemKind.Folder,
-				isDirectory: true,
-				isFile: false,
-				replacementIndex: 3,
-				replacementLength: 2 // replacing ./
-			}]);
-		});
-		test('cd ./f|', async () => {
+			const result = await terminalCompletionService.resolveResources(resourceRequestConfig, './h', 3, provider);
```

Comment:
We should make it so we pass in the Windows state here, since this is called on the frontend. Maybe via something like `resourceRequestConfig.secondaryPathSeparators: string[]`?

Reply:
how about replacing all / or \ in the prefix with the pathSeparator?

Reply:
done

Reply:
hm, i'm thinking that's less efficient, so will do the secondaryPathSeparators idea or similar instead